### ISSUE TITLE
MobX 6.0.x

### ISSUE THREAD

**Issue by DmitryTevtonsky** [quarrant/mobx-persist-store#9]
I appreciate your library.

How i can implement you library with mobx 6.0.x?

There some new features.
The decorate API has been removed and now it implement with makeObservable and makeAutoObservable.
[changelog-mobx](https://changelogs.xyz/mobx)

Thank you.

**Comment by quarrant** [comment]
Good question! I added support mobx6. Try to use version 0.5.13 like https://github.com/quarrant/mobx-persist-store#with-mobx-6

**Comment by MaRaSu** [comment]
Mobx-persist-store indeed seems to be working well with mobx 6.1.4, however when installing it you have to give --legacy-peer-deps to npm to override peerDependencies --> worth fixing perhaps?

I have a setup with few mobx stores, combined in a RootStore (which itself is not observable), when using mobx-persist-store persistence method inside my RootStore's constructor mobx (in most strict mode config) complains about accessing observables outside of a reactive context --> had to wrap call to persistence with runInAction to dismiss the mobx warning.

**Comment by deadcoder0904** [comment]
@quarrant would love for you to fix peer deps as well & release a new version. i can't install `mobx-persist-store` as it throws this error:

```bash
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: myapp@1.0.0
npm ERR! Found: mobx@6.1.5
npm ERR! node_modules/mobx
npm ERR!   mobx@"^6.1.5" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer mobx@"^5.14.0" from mobx-persist-store@0.5.15
npm ERR! node_modules/mobx-persist-store
npm ERR!   mobx-persist-store@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/deadcoder0904/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/deadcoder0904/.npm/_logs/2021-02-08T11_29_07_702Z-debug.log
```

**Comment by quarrant** [comment]
@MaRaSu can you show me your code ?

**Comment by MaRaSu** [comment]
@quarrant, sure, this is what RootStore looks like where I had to add that extra runInAction:

```
import { persistence, StorageAdapter } from 'mobx-persist-store';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { UIStore } from './UIStore';
import { GeofeaturesStore } from './GeofeaturesStore';
import { DeviceStore } from './DeviceStore';

function readStore(name) {
	return new Promise(async (resolve) => {
		const data = await AsyncStorage.getItem(name);
		resolve(JSON.parse(data));
	});
}

function writeStore(name, content) {
	return new Promise((resolve) => {
		AsyncStorage.setItem(name, JSON.stringify(content));
		resolve();
	});
}

export class RootStore {
	uiStore
	geofeaturesStore
	deviceStore

	constructor() {
		runInAction(() => {
			this.uiStore = persistence({
				name: 'UIStore',
				properties: [
					'mapBaseLayerId',
					'mapviewRegion'
				],
				adapter: new StorageAdapter({
					read: readStore,
					write: writeStore,
				}),
				reactionOptions: {
					// optional
					delay: 2000
				},
			})(new UIStore(this));
		});
		this.geofeaturesStore = new GeofeaturesStore(this);
		this.deviceStore = new DeviceStore(this);
	}
}

export const rootStore = new RootStore();
export const rootStoreContext = createContext();
```

**Comment by quarrant** [comment]
@MaRaSu does the problem still exist? After fix #21
